Basket-Donation-Payments/Basket-Donation-Payments/View Controllers/Login & Signup Flow/UserSignupViewController.m

```diff
@@ -32,6 +32,18 @@ - (void)viewDidLoad {
     [super viewDidLoad];
     self.usernameTextField.text = self.username;
     self.passwordTextField.text = self.password;
+    self.firstNameTextField.delegate = self;
+    self.lastNameTextField.delegate = self;
+    self.emailTextField.delegate = self;
+    self.usernameTextField.delegate = self;
+    self.passwordTextField.delegate = self;
+    UITapGestureRecognizer *tap = [[UITapGestureRecognizer alloc] initWithTarget:self.view action:@selector(endEditing:)];
+    [self.view addGestureRecognizer:tap];
+}
+
+- (BOOL)textFieldShouldReturn:(UITextField *)textField {
+    [textField resignFirstResponder];
```

Comment:
Just an idea to improve the UX if you have any time left at the end, you can add all text fields to an array in the order they're rendered on the screen, and here you can do `[textFields[i] resignFirstResponder];` and `[textFields[i+1] becomeFirstResponder];`, so you can fill all the form just by tapping next on the keyboard.